src/coreclr/jit/llvmdebuginfo.cpp

```diff
@@ -234,9 +234,11 @@ void Llvm::declareDebugVariables()
     DILocation* debugLocation = getArtificialDebugLocation();
     Instruction* insertInst = _builder.GetInsertBlock()->getTerminator();
     Value* spilledShadowStackAddr = nullptr;
-    for (auto lcl = m_debugVariablesMap.Begin(); !lcl.Equal(m_debugVariablesMap.End()); ++lcl)
+    for (auto lcl :
+         JitHashTable<unsigned, JitSmallPrimitiveKeyFuncs<unsigned>, llvm::DILocalVariable*>::KeyValueIteration(
+             &m_debugVariablesMap))
```

Comment:
```suggestion
    for (auto lcl : decltype(m_debugVariablesMap)::KeyValueIteration(&m_debugVariablesMap))
```
`decltype` can be useful in situations like this.